test/integration/instance.test.js

```diff
@@ -2076,6 +2076,69 @@ describe(Support.getTestDialectTeaser('Instance'), () => {
     });
   });
 
+  describe('isSoftDeleted', () => {
+    beforeEach(function() {
+      this.ParanoidUser = this.sequelize.define('ParanoidUser', {
+        username: { type: DataTypes.STRING }
+      }, { paranoid: true });
+
+      this.ParanoidUser.hasOne(this.ParanoidUser);
```

Comment:
I'm not entirely sure but I believe so. https://github.com/sequelize/sequelize/pull/7531/files/354210b1992afccaae8fdb270fddcb8e4752923f#diff-2756a0ac12fc81f7843fd569393b5966R1655 has the same code and I found that I needed that logic here.